DNNConnect.CKEditorProvider.dnn

```diff
@@ -84,7 +84,7 @@
             <install>
               <configuration>
                 <nodes>
-                  <node path="/configuration/dotnetnuke/htmlEditor[@defaultProvider='']" action="updateattribute" name="defaultProvider" value="DNNConnect.CKE" />
+                  <node path="/configuration/dotnetnuke/htmlEditor[@defaultProvider]" action="updateattribute" name="defaultProvider" value="DNNConnect.CKE" />
```

Comment:
Looking in the history of this file (commit 3b684c3db671f0fc1ca9742c724f47479fbc0576), the initial intention was to set CKEditor as the default HTML provider **only** if there was no other default provider already set. This new XPath expression will set CKEditor as the default **even** if there is some other provider already set.

Also, the `defaultProvider` attribute is mandatory (DNN crashes if absent), so specifying just the attribute in the XPath without a value criteria is redundant. It's the same as `/configuration/dotnetnuke/htmlEditor`.

So if the intention is now to set CKEditor as the default, despite of any existing selection, then I think the expression is correct but could just be `/configuration/dotnetnuke/htmlEditor`

Reply:
This issue does not happen if CKEditor is not the default HTML provider at the time of uninstall. When CKEditor is the default, uninstall process changes the default to "DotNetNuke.RadEditorProvider". Later on, when reinstalling, this default is not changed, and that creates the reported error. Manually changing the `defaultProvider` attribute to "DNNConnect.CKE" does the trick.
Maybe, the correct way to fix this is to set DNNConnect.CKE as the default during installation only if no other one is specified, or if the default is DotNetNuke.RadEditorProvider.

Reply:
👍 sounds good to me, match on `[@defaultProvider=""]` and `[@defaultProvider="DotNetNuke.RadEditorProvider"]`

Reply:
Cool. So @Mhtshum the XPath should be
```
/configuration/dotnetnuke/htmlEditor[@defaultProvider='' or @defaultProvider='DotNetNuke.RadEditorProvider']
```

Reply:
@daguiler  and @bdukes 
Following selector won't work for install as both depends upon value of attribute, which is currently the case is
/configuration/dotnetnuke/htmlEditor[@defaultProvider='' or @defaultProvider='DotNetNuke.RadEditorProvider']

What I'm understanding is that currently Installing is configured to make the package default HTML PROVIDER installer, hence the targeting the htmlProvider either 
- by attribute(more performant) 
or
- by node before this [commit](https://github.com/DNN-Connect/CKEditorProvider/commit/3b684c3db671f0fc1ca9742c724f47479fbc0576)  

both are OK (which is currently fixing this issue)

Reply:
Sorry @Mohtshum, I can't follow your comment above

Reply:
Currently selector is :
- /configuration/dotnetnuke/htmlEditor[@defaultProvider='']
And installation is failing.
 
- /configuration/dotnetnuke/htmlEditor[@defaultProvider='DotNetNuke.RadEditorProvider']
this will only work if value matches with DotNetNuke.RadEditorProvider and if some different value then it would also not work.

While 
/configuration/dotnetnuke/htmlEditor[@defaultProvider]
this will target the node regardless the value of defaultProvider attribute upon installation and make CKEditor as default html provider.

Reply:
and what about the XPath query I proposed?

Reply:
@Mhtshum we're saying that installing the provider _should not_ make it the default, unless :one: there is no default, or :two: the old default is in place.  If there's is a different editor in use, installing this package should not change that default.

Reply:
@daguiler  that is also OK as it is before [commit](https://github.com/DNN-Connect/CKEditorProvider/commit/3b684c3db671f0fc1ca9742c724f47479fbc0576)

Reply:
> as it is before commit

I don't understand @Mhtshum. Never in the history of the file the XPath was
`/configuration/dotnetnuke/htmlEditor[@defaultProvider='' or @defaultProvider='DotNetNuke.RadEditorProvider']`
I just created a suggestion so it's more evident the change we're requesting.